### DISCUSSION TITLE
CKS32 clone appears working but doesn't

### DISCUSSION THREAD

**Discussion by lausvi** [erichelgeson/BlueSCSI-v1#164] [General]
Hi,

while assembling couple more BlueSCSIs, I ordered BluePills from eBay and didn't pay enough attention and bought CKS32 clones (CKS32F103C8T6, it was mentioned in the ad...) instead of "real deals" (or "better clones").

I assembled one external unit (and naturally soldered it in) and everything appeared to be working fine; flashing went fine and I was able to insert my previously used SD card with a Mac partitioned disk and it appeared on my Centris 610's desktop. HOWEVER once I tried to launch any Mac apps that were on that drive, I would get all sorts of errors: type 3, type 13 (MacOS 8) or even a bomb-System error "Finder - bus error" (on my OS7 IIcx). I was able to copy files and read fe. smaller text files but none of the apps worked. SCSI utilities were able to test the partition etc. 

I tried another SD card (not Sandisk this time, as I found some suggestions that the CKS32 would have problems with Sandisks) and new partitions but there would be all sort of issues. 

As an internal drive I was able to install MacOS, but the first reboot would only halt the machine (or bomb out with again various bus/illegal instruction errors). Pre-made image would hang booting.

I bought only a couple of these, so no real harm done but I did spent some time trying to figure out what was wrong. 😅 

Is there anything I could try to make these work, like any of the alternative firmwares or such? 

Are there currently any known-good eBay stores for buying new BluePills?

**Comment by erichelgeson** [comment]
I've used many CKS32's in the past - they actually preform sightly better and work well. The SanDisk Issue you mention is related to CH32 clones. I would go back through the troubleshooting guide, reformat low level with SD Card Formatter.

**Reply by lausvi** [reply]
I did another test: I used a new card (Transcend) and used SD Card Formatter (tested both FAT32 as made by the SD Card Formatter and then re-formatted one to ExFAT). 

I copied the old image file to it and used a CH32 on an external BlueSCSI on my Mac IIcx. At first I was able to launch apps from the drive, but after a while they started failing. For some reason the drive gets corrupted. I formatted the drive on the Mac, and copied some stuff from the machine (that works) to it and they failed to run from the external drive again.

This is the log from the card:
VERSION: 1.1-20220626-USB
DEBUG:0 SDFAT_FILE_TYPE:3
SdFat version: 2.1.2
Sd Format: exFAT
SPI speed: 50Mhz
SdFat Max FileName Length: 32
Initialized SD Card - lets go!
Sd MID:74 OID:J`
Sd Name:USDU1
Sd Date:3/2016
Sd Serial:267715141

Edit: update: same card + same image + same board: always works, and always breaks if I replace the SMT32 with the CKS32.
